packages/npm-dependencies-validation/src/utils/packageJsonUtil.ts

```diff
@@ -0,0 +1,116 @@
+import { access, readFile } from "fs/promises";
+import { constants } from "fs";
+import { join, dirname, sep, relative } from "path";
+import { VscodeFsUri, VscodeWsFolder } from "../types";
+
+const yarnManagerFiles = [
+  "yarn.lock",
+  ".yarnrc",
+  ".yarnrc.yml",
+  ".yarn",
+  "./node_modules/.yarn-integrity",
+];
+const pnpmManagerFiles = [
+  "pnpm-workspace.yaml",
+  "pnpm-lock.yaml",
+  ".pnpmfile.cjs",
+  "./node_modules/.pnpm",
+];
+const npm7Properties = ["workspaces"];
+
+async function readJsonFile(packageJsonUri: VscodeFsUri): Promise<any> {
+  try {
+    const packageJsonContent = await readFile(packageJsonUri.fsPath, "utf-8");
+    const content: { name: string } = JSON.parse(packageJsonContent);
+    return content;
+  } catch (error: any) {
+    return {};
+  }
+}
+
+async function isPathExist(packageJsonPath: string): Promise<boolean> {
```

Comment:
`isPathExist` --> `isFileExist`?

Reply:
`packageJsonPath` arg --> absPath (assuming it must be absolute)

Reply:
It is used not just for files.
Changed to absPath